staging/src/k8s.io/component-helpers/node/util/ips.go

```diff
@@ -65,18 +66,18 @@ func parseNodeIP(nodeIP string, allowDual, sloppy bool) ([]net.IP, error) {
 // ParseNodeIPArgument parses kubelet's --node-ip argument. If nodeIP contains invalid
 // values, they will be logged and ignored. Dual-stack node IPs are allowed if
 // cloudProvider is unset, or if it is `"external"` and allowCloudDualStack is true.
-func ParseNodeIPArgument(nodeIP, cloudProvider string, allowCloudDualStack bool) ([]net.IP, error) {
+func ParseNodeIPArgument(ctx context.Context, nodeIP, cloudProvider string, allowCloudDualStack bool) ([]net.IP, error) {
```

Comment:
Requiring a context when only a logger is needed makes it harder to call this function in code which only has a logger. I prefer passing a logger here.